cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -854,13 +854,6 @@ else()
   set(THRIFT_REQUIRES_BOOST FALSE)
 endif()
 
-# Parquet requires boost only with gcc 4.8 (because of missing std::regex).
-if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS "4.9")
-  set(PARQUET_REQUIRES_BOOST TRUE)
-else()
-  set(PARQUET_REQUIRES_BOOST FALSE)
-endif()
-
```

Comment:
Should also remove the lines with `BOOST_REGEX_LIBRARY` below, right? Nothing else uses boost::regex in our code. 

We can also remove regex.hpp from cpp/build-support/trim-boost.sh. That should probably be a followup JIRA, but could you edit the comment on L38 of that script to say that we actually don't need it anymore?

Reply:
Thanks.
I didn't notice that we need boost::regex only for this feature. I've removed boost::regex dependency entirely.